sdk/src/lib.rs

```diff
@@ -110,6 +110,11 @@ impl ProverClient {
             .as_ref()
             .ok_or_else(|| anyhow::anyhow!("Network client not initialized"))?;
 
+        // Execute the runtime before creating the proof request.
+        let mut runtime = Runtime::new(Program::from(elf));
+        runtime.write_vecs(&stdin.buffer);
+        runtime.run();
```

Comment:
maybe we should log `simulation complete` or something, that way people have a better understanding if this does take a significant period of time